### ISSUE TITLE
False positive unused-variable when using locals()

### ISSUE THREAD

**Issue by danijar** [PyCQA/pylint#1112]
```python
def example():
    one = 13
    two = 42
    print('{one} {two}'.format(**locals()))
```

```sh
$ python -m pylint -r no example.py
No config file found, using default configuration
************* Module test
C:  1, 0: Missing module docstring (missing-docstring)
C:  1, 0: Missing function docstring (missing-docstring)
W:  3, 4: Unused variable 'two' (unused-variable)
W:  2, 4: Unused variable 'one' (unused-variable)
```

**Comment by moylop260** [comment]
+1

**Comment by nait-aul** [comment]
+1

\+ [PEP 498](https://www.python.org/dev/peps/pep-0498/) for new string formatting in Python 3.6 like:

```python
>>> x = "Hello"
>>> y = "World!"
>>> f"{x} {y}"
Hello World!
```

**Comment by PCManticore** [comment]
The issue with locals() won't probably be fixed. The second one with f-strings will probably land once we support 3.6, which is not the case just yet.

**Comment by danijar** [comment]
@PCManticore What's the reasoning to not fix the locals() issue?

**Comment by The-Compiler** [comment]
@danijar https://github.com/PyCQA/pylint/issues/641#issuecomment-163178303 has some thoughts on it

**Comment by danijar** [comment]
Thanks for pointing me to that issue. Closing this as duplicate. We should create a new issue for the new-style string formatting since it's not related to `locals()`.
